application/views.py

```diff
@@ -100,34 +106,33 @@ def save_existing_application(request, application_id):
             )
     else:
         form = HighSchoolApplicationForm()
-    context = {"form": form, "application_id": application_id}
+        f = None
+    context = {"form": form, "application_id": application_id, "selected_app": f}
     return render(request, "application/index.html", context)
 
 
 def all_applications(request):
-    user_type = request.session.get("user_type", None)
-    username = request.session.get("username", None)
-    if (
-        not request.session.get("is_login", None)
-        or not username
-        or user_type != UserType.STUDENT
-    ):
+    username = check_current_session(request)
+    if not username:
         return redirect("landingpage:index")
     user = Student.objects.get(username=username)
-    context = {"applications": HighSchoolApplication.objects.filter(user_id=user.pk)}
+    context = {
+        "applications": HighSchoolApplication.objects.filter(user_id=user.pk).order_by(
+            "-is_draft", "-submitted_date"
+        )
+    }
     return render(request, "application/index.html", context)
 
 
 def detail(request, application_id):
-    user_type = request.session.get("user_type", None)
-    username = request.session.get("username", None)
-    if (
-        not request.session.get("is_login", None)
-        or not username
-        or user_type != UserType.STUDENT
-    ):
+    username = check_current_session(request)
+    if not username:
         return redirect("landingpage:index")
-    application = HighSchoolApplication.objects.get(pk=application_id)
+    try:
+        application = HighSchoolApplication.objects.get(pk=application_id)
+    except Exception:
+        context = {"invalid_url_app": "Application not found."}
```

Comment:
Handling this using the broken.html pages would increase the consistency across our website